src/Spectre.libGaClassifier/IndividualFeasibilityConditionsFactory.cpp

```diff
@@ -0,0 +1,95 @@
+#include "IndividualFeasibilityConditionsFactory.h"
+#include "Spectre.libGenetic/LengthCondition.h"
+#include "Spectre.libGenetic/MinimalFillupCondition.h"
+#include "Spectre.libGenetic/MaximalFillupCondition.h"
+#include "Spectre.libGenetic/MinimalPercentageFillupCondition.h"
+#include "Spectre.libGenetic/MaximalPercentageFillupCondition.h"
+#include "AllLabelTypesIncludedCondition.h"
+#include "Spectre.libException/ArgumentOutOfRangeException.h"
+#include "Spectre.libGenetic/InconsistentMinimalAndMaximalFillupException.h"
+
+namespace spectre::supervised
+{
+IndividualFeasibilityConditionsFactory::IndividualFeasibilityConditionsFactory(gsl::span<const supervised::Label> labels, size_t length, size_t minimalFillup, size_t maximalFillup):
+    m_Length(length),
+    m_MinimalFillup(minimalFillup),
+    m_MaximalFillup(maximalFillup),
+    m_MinimalPercentageFillup(0),
+    m_MaximalPercentageFillup(1),
+    m_Labels(labels)
+{
+    if (m_MinimalFillup > m_MaximalFillup)
+    {
+        throw algorithm::genetic::InconsistentMinimalAndMaximalFillupException(m_MinimalFillup, m_MaximalFillup);
+    }
+}
+
+IndividualFeasibilityConditionsFactory::IndividualFeasibilityConditionsFactory(gsl::span<const supervised::Label> labels, size_t length, double minimalPercentageFillup, double maximalPercentageFillup):
+    m_Length(length),
+    m_MinimalFillup(0),
+    m_MaximalFillup(std::numeric_limits<size_t>::max()),
+    m_MinimalPercentageFillup(minimalPercentageFillup),
+    m_MaximalPercentageFillup(maximalPercentageFillup),
+    m_Labels(labels)
+{
+    if (m_MinimalPercentageFillup > m_MaximalPercentageFillup)
+    {
+        throw algorithm::genetic::InconsistentMinimalAndMaximalFillupException(m_MinimalPercentageFillup, m_MaximalPercentageFillup);
+    }
+    if (m_MinimalPercentageFillup < 0 || minimalPercentageFillup > 1)
+    {
+        throw core::exception::ArgumentOutOfRangeException<double>("minimalPercentageFillup", 0, 1, m_MinimalPercentageFillup);
+    }
+    if (m_MaximalPercentageFillup < 0 || m_MaximalPercentageFillup > 1)
+    {
+        throw core::exception::ArgumentOutOfRangeException<double>("maximalPercentageFillup", 0, 1, m_MaximalPercentageFillup);
+    }
+}
+
+std::unique_ptr<algorithm::genetic::BaseIndividualFeasibilityCondition> IndividualFeasibilityConditionsFactory::build() const
+{
+    std::unique_ptr<algorithm::genetic::BaseIndividualFeasibilityCondition> condition = nullptr;
+    if (m_Length != 0)
+    {
+        condition = std::make_unique<algorithm::genetic::LengthCondition>(m_Length, std::move(condition));
+    }
+    if (m_MinimalFillup != 0)
+    {
+        condition = std::make_unique<algorithm::genetic::MinimalFillupCondition>(m_MinimalFillup, std::move(condition));
+    }
+    if (m_MaximalFillup != std::numeric_limits<size_t>::max())
+    {
+        condition = std::make_unique<algorithm::genetic::MaximalFillupCondition>(m_MaximalFillup, std::move(condition));
+    }
+    if (m_MinimalPercentageFillup != NAN)
```

Comment:
Do you know, that `NAN` is so special, that it's actually *inequal to itself*? Try [`std::isnan`](http://en.cppreference.com/w/cpp/numeric/math/isnan).

Reply:
seriously? nice, done

Reply:
Check [this](https://en.wikipedia.org/wiki/NaN#Comparison_with_NaN) for behavior of equality operator :)